test/Tester/DependencyInjectionGrainTests.cs

```diff
@@ -39,7 +41,49 @@ protected override TestCluster CreateTestCluster()
 
             // the injected service will return the same value only if it's the same instance
             Assert.Equal(
-                await grain1.GetStringValue(), 
+                await grain1.GetStringValue(),
+                await grain2.GetStringValue());
+        }
+
+        [Fact, TestCategory("BVT"), TestCategory("Functional")]
+        public async Task CannotGetExplictlyRegisteredGrain()
+        {
+            ISimpleDIGrain grain = GrainFactory.GetGrain<ISimpleDIGrain>(GetRandomGrainId(), grainClassNamePrefix: "UnitTests.Grains.ExplicitlyRegistered");
+            var exception = await Assert.ThrowsAsync<OrleansException>(() => grain.GetTicksFromService());
+            Assert.Contains("Error creating activation for", exception.Message);
+            Assert.Contains(nameof(ExplicitlyRegisteredSimpleDIGrain), exception.Message);
+        }
+    }
+
+    public class DelegateDependencyInjectionGrainTests : OrleansTestingBase, IClassFixture<DelegateDependencyInjectionGrainTests.Fixture>
+    {
+        private class Fixture : BaseTestClusterFixture
+        {
+            protected override TestCluster CreateTestCluster()
+            {
+                return new TestCluster(new TestClusterOptions(1))
+                {
+                    AppDomainSiloCreatorType = typeof(SiloCreator)
```

Comment:
If we get rid of stateful statics and can just use the container instead, then we don't need IPC or AppDomains